module/VuFind/src/VuFind/ILS/Logic/Holds.php

```diff
@@ -510,9 +534,11 @@ protected function getRequestDetails($details, $HMACKeys, $action)
 
         // Build Params
         return [
-            'action' => $action, 'record' => $details['id'],
-            'source' => $details['source'] ?? DEFAULT_SEARCH_BACKEND,
-            'query' => $queryString, 'anchor' => '#tabnav',
+            'action' => $action,
+            'record' => $linkOverrides['id'] ?? $details['id'],
+            'source' => $linkOverrides['source'] ?? ($details['source'] ?? DEFAULT_SEARCH_BACKEND),
```

Comment:
Not sure if these parentheses are mine or yours, but in any case, I'm pretty sure we don't need them.

```suggestion
            'source' => $linkOverrides['source'] ?? $details['source'] ?? DEFAULT_SEARCH_BACKEND,
```